### PR TITLE
Task 3148: Create Typescript view models automatically (from JSON api)

### PR DESCRIPTION
**Pull request by bcalvignac**
Picked the swagger-ts-generator package.
I rejected the following packages: swagger-angular-generator, ng-swagger-gen, swagger-codegen and NSwag.CodeGeneration because their services generation was not helpful (i.e function parameters were wrapped in objects)

Also remove yarn

### CODE REVIEW

File: generateViewModels.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+var https = require('https');
+var fs = require('fs');
+var path = require('path');
+const { generateTSFiles } = require("swagger-ts-generator");
+
+var url = "https://dev.api.news.gov.bc.ca/swagger/v1/swagger.json";
```
Comment (brianorwhatever) [inline generateViewModels.js]: I don't think we should hard code the URL here

**Comment by brianorwhatever** [comment]
There is a lot of new models here. Will they all be used in the dashboard project? I am hesitant to say we should include all of them if they aren't being used. Will these effect the size of the final JS files that are generated? I believe angular does a lot of dropping unused code but I would want to know for sure.
There is also now 2 classes named "Theme" since we're importing the old one.. should I rename the front end model for the new themes to be messages as I did on the backend?

File: generateViewModels.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+var https = require('https');
+var fs = require('fs');
+var path = require('path');
+var request = require('request');
+var jsYaml = require('js-yaml');
+var unzip = require('unzip2');
+
+var newsApiUrl = "https://dev.api.news.gov.bc.ca/swagger/v1/swagger.json";
+var hubApiUrl = "https://dev.api.hub.gov.bc.ca/swagger/v1/swagger.json";
```
Comment (brianorwhatever) [inline generateViewModels.js]: should be in a config file

File: generateViewModels.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  https.get(url, function (response) {
+    response.pipe(file);
+    file.on('finish', function () {
+      file.close(downloadCB(tsPath, models2get));  // close() is async, call downloadCB after close completes.
+    });
+  });
+}
+
+download(newsApiUrl, "/news", ["/dataIndex", "/dataModel", "/post", "/document", "/asset", "/keyValuePair", "/category", "/ministr", "/minister", "/contact", "/resourceLink"]);
+//downloadCB("./src/app/view-models/hub");
```
Comment (brianorwhatever) [inline generateViewModels.js]: remove commented out line

File: generateViewModels.js (hunk 2 above)
Comment (brianorwhatever) [inline generateViewModels.js]: Is the hub api going to be public? Won't it need to be authed to build these models

File: generateViewModels.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  var file = fs.createWriteStream(swaggerTempFile);
+  https.get(url, function (response) {
+    response.pipe(file);
+    file.on('finish', function () {
+      file.close(downloadCB(tsPath, models2get));  // close() is async, call downloadCB after close completes.
+    });
+  });
+}
+
+download(newsApiUrl, "/news", ["/dataIndex", "/dataModel", "/post", "/document", "/asset", "/keyValuePair", "/category", "/ministr", "/minister", "/contact", "/resourceLink"]);
```
Comment (brianorwhatever) [inline generateViewModels.js]: what is /ministr?

File: package.json [hunk 5]
```diff
@@ -31,6 +32,7 @@
     "font-awesome": "^4.7.0",
     "npm": "^6.4.1",
     "rxjs": "~6.3.3",
+    "rxjs-compat": "^6.3.3",
```
Comment (brianorwhatever) [inline package.json]: What did you need this for? I don't see it being referenced anywhere

File: generateViewModels.js (hunk 2 above)
Comment (bcalvignac) [reply generateViewModels.js]: Yes, but we will already have the auth key in the Dashboard

**Pull request merged**